marklogic-data-hub/src/test/java/com/marklogic/hub/job/JobDocManagerTest.java

```diff
@@ -32,6 +32,24 @@ public void testGetLatestJobDocumentForFlows() {
         Assertions.assertEquals("newestJob", latestJobsForFlow.get("job").get("jobId").asText(), "The latest JobId for 'hub' should be 'newestJob'");
     }
 
+    @Test
+    public void testJobTimesForSameJobId() {
```

Comment:
For this test, we really need to run a flow, not just make arbitrary calls to create a job. 

To do that, you'll need to install a project first; there are a bunch of them under src/test/resources/test-projects. 

A good simple one is "simple-customer-flow":

```
installProjectInFolder("test-projects/simple-customer-flow");
```

You can then run a flow like this, which runs step "1" (and you don't need it to process any data:

runFlow(new FlowInputs("simpleCustomerFlow", "1").withJobId("abc123"));

After running the flow, you can use jobDocManager to get the job doc and grab the times on it. Then you can run the flow again with the same jobId, and compare the times.

Also, the test is difficult to read right now because there's no separation between the lines of code. Think about the steps in your test here:

- Run a flow
- Get the job times
- Run the flow again
- Get the job times again
- Compare the job times

So you have 5 steps to your test - that suggests that 5 blocks of code (i.e. with blank lines in between them) is a good starting point for making the test easier to read.

Reply:
Also, let's put the test into com.marklogic.hub.flow - make a new test class called "RunFlowWithSameJobIdTest", as that's the exact thing we're testing.

Reply:
Thank you Rob for the comments and detailed guidance for the test to be added.
I have added test RunFlowWithSameJobIdTest as mentioned.